app/server/api/filter_api.py

```diff
@@ -52,7 +52,7 @@ def post(self):
         create_filter = SavedFilter(
             name=filter_name,
             filter=filter_attributes,
-            organisation_id=g.user.default_organisation_id
```

Comment:
This was causing all filters to be assigned to a users default organisation (even if they were on another active organisation)